misc.Rmd

```diff
@@ -2,56 +2,71 @@
+    We regard demos as a legacy phenomenon, whose goals are now better met by vignettes (@sec-vignettes).
+    For actively maintained packages, it probably makes sense to repurpose the content in any existing demos somewhere that's more visible, e.g. in `README` or in vignettes.
+    These other locations offer other advantages, such as making sure that the code is exercised regularly.
+    This is not true of actual demos, leaving them vulnerable to rot.
 
 -   `exec/`: for executable scripts.
     Unlike files placed in other directories, files in `exec/` are automatically flagged as executable.
+    Empirically, to the extent that R package are shipping scripts for external interpreters, the `inst/` directory seems to be preferred location these days.
 
 -   `po/`: translations for messages.
-    This is useful but beyond the scope of this book.
-    See the [Internationalization](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Internationalization) chapter of "R extensions" for more details.
-
--   `tools/`: auxiliary files needed during configuration, or for sources that need to generate scripts.
+    This is useful, but beyond the scope of this book.
+    See the [Internationalization](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Internationalization) chapter of "Writing R extensions" for more details.
 
 ## Installed files {#inst}
 
-When a package is installed, everything in `inst/` is copied into the top-level package directory.
-In some sense `inst/` is the opposite of `.Rbuildignore` - where `.Rbuildignore` lets you remove arbitrary files and directories from the top level, `inst/` lets you add them.
-You are free to put anything you like in `inst/` with one caution: because `inst/` is copied into the top-level directory, you should never use a subdirectory with the same name as an existing directory.
-This means that you should avoid `inst/build`, `inst/data`, `inst/demo`, `inst/exec`, `inst/help`, `inst/html`, `inst/inst`, `inst/libs`, `inst/Meta`, `inst/man`, `inst/po`, `inst/R`, `inst/src`, `inst/tests`, `inst/tools` and `inst/vignettes`.
+When a package is installed, everything in `inst/` is copied into the top-level directory of the installed package.
+In some sense `inst/` is the opposite of `.Rbuildignore` - where `.Rbuildignore` lets you remove arbitrary files and directories from the built package, `inst/` lets you add them.
 
-This chapter discusses the most common files found in `inst/`:
+You are free to put anything you like in `inst/` with one caution: because `inst/` is copied into the top-level directory, don't create a subdirectory that collides with any of the directories that make up the official structure of an R package.
+We recommend avoiding directories with special significance in either the source or installed form of a package, such as: `inst/data`, `inst/help`, `inst/html`, `inst/libs`, `inst/man`, `inst/Meta`, `inst/R`, `inst/src`, `inst/tests`, `inst/tools`, and `inst/vignettes`.
```

Comment:
I'm still listing `inst/tools` here as something we discourage, even though you'll see it in some tidyverse / r-lib packages. I think in all / most cases, those would be better as an `.Rbuildignored` top-level `tools/` directory. Or, better still, some generalization of `data-raw/`. But I know we're not necessarily getting into that at this point.